Yoast/javascript
# LINGO-548 Merge Hungarian feature branch

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Edits typos and add more transition words for Hungarian.
* [yoastseo] Adds function words for Hungarian.
* [yoastseo] Adds the sentence beginning assessment and specs for Hungarian.
* [yoastseo] Adds specs for the sentence length assessment in Hungarian.
* [yoastseo] Implements Hungarian stemmer.
* [Yoast SEO Free] Improves keyphrase recognition in Hungarian by filtering the function words such as harmadik, tiétek, valaki.
* [Yoast SEO Free] Improves the transition words assessment for Hungarian.
* [Yoast SEO Free] Implements the consecutive sentences assessment for Hungarian to ensure variety in a text.
* [Yoast SEO Premium] Introduces word form support for Hungarian.


## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Set your website to Hungarian

   * In Yoast Seo Free:

      i. For the sentence beginning assessment:
      * Create a new post and add three Hungarian sentences that all begin with the same written-out number, like _egy_ ("one"). E.g. `Egy macska. Egy kutya. Egy teknős`. The consecutive sentences readability assessment should read: `Consecutive sentences: There is enough variety in your sentences. That's great!`
      * Replace the sentences with three sentences that all begin with the same demonstrative pronoun, like _ez_ ("this"). E.g. `Ez a ház csinos. Ez az ajtó nagy. Ez az erkély nagyszerű kilátást nyújt`. The consecutive sentences assessment should again read: `Consecutive sentences: There is enough variety in your sentences. That's great!`
     * Replace the sentences with the following three sentences: `Reggel van. Reggel álmos vagyok. Reggel dolgozni kell`. The consecutive sentences assessment should read: `Consecutive sentences: The text contains 3 consecutive sentences starting with the same word. Try to mix things up!`                 
     
     ii. For the sentence length assessment:
     * Paste a Hungarian sentence with 20 words three times in the text. You can use for example the following sentence.:
`A magyar nyelv az uráli nyelvcsalád tagja, a finnugor nyelvek közé tartozó ugor nyelvek egyike legközelebbi rokonai a manysi és.`
     * Make sure that your sentence length assessment score reads: `Sentence length: Great!`
     * Make all of your sentence longer than 20 words (for example by repeating some words).
     * Check that the Sentence length assessment score reads: `Sentence length: 100% of the sentences contain more than 20 words, which is more than the recommended maximum of 25%. Try to shorten the sentences.`

     iii. For the transition word assessment:
     * Paste the following text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce dictum semper tempus. Nulla fermentum molestie odio, ac imperdiet nunc venenatis vel. Praesent feugiat velit ac est aliquam, ac maximus lobortis.`
     * Add a single transition word to each sentence. Example of single transition words: `semmint, vajon, ugye, miszerint, nemde`.
     * The transition words assessment should read `Well done!` and the sentences containing transition words should be highlighted with the eye marker.
     * Delete transition words from each sentence apart from one.
     * The transition word assessment should read: `Only 25% of the sentences contain transition words, which is not enough. Use more of them.`.
     * Repeat the previous steps using multi-word transition words, e.g.: `így tehát`; `ilyen körülmények között`; `késedelem nélkül`; `kétség nélkül`; `más szóval`.
     * Repeat the previous steps using two-part transition words, .eg.: `nem úgy... mint`; `oly módon... hogy`; `inkább... semhogy`. The `...` should be replaced with words from the sentences, e.g. Lorem ipsum dolor sit amet **nem úgy**, consectetur adipiscing **mint** elit.

     iv. To test function words:
     * Create a new post and add a text with at least 300 words, e.g. using the [lorem ipsum generator](https://www.lipsum.com/).
     * Set your keyphrase to `tizenkilenc macska`
     * Add the word `macska` to the text
     * Make sure that the keyphrase density shows that the keyphrase is found once. 


* In Yoast SEO Premium (for the QA team) or in the content-analysis app (for the lingo team, since morphology cannot be tested in the plugin before the RC is created):
   * Change site language to Hungarian.
   * Add a Hungarian text of at least 300 words. (You can use Wikipedia as a source)
   * Set `mindennapjaim` as keyphrase and enter `mindennap` in the text/meta description/title. The relevant assessments should recognise `mindennap` as an instance of the keyphrase. You should see the following assessment feedback:
      * `Keyphrase density: The focus keyphrase was found 1 time. That's less than the recommended minimum of 3 times for a text of this length. Focus on your keyphrase!` (Number depending on how often you added the keyphrase in the text)
      * `Keyphrase in title: Does not contain the exact match. Try to write the exact match of your keyphrase in the SEO title and put it at the beginning of the title.`
      * `Keyphrase in meta description: Keyphrase or synonym appear in the meta description. Well done!`
   * Switch the keyphrase and the form in the text/meta description/title around and make sure the assessment results are still 
     the same.
   * Repeat the same process with `más` as keyphrase and `másoké` in the text/meta description/title.


## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-548
